### PR TITLE
Convert links to https (where supported) in docs/standard

### PR DESCRIPTION
**Pull request by ChrisMaddock**
## Summary

Relates to #4504. This changes all the links in `docs/standard` to use https - where a) https is supported, and b) the link does not redirect.

As discussed on this issue, I wrote a small tool to do this. The tools, currently a little overkill - but I want to use it for more-involved tasks in future. 

The code can be seen [here](https://github.com/ChrisMaddock/https-checker). There's two classes of interest:

**HttpsChecker**: Takes a link, try's the https variant of it. Success if https returns 200, fails if anything else. This includes broken links, and links that don't support https. It also fails for redirects, as it really wants manual validation that the link is redirecting to an appropriate place - I have different plans to tackle some of the main cases there in future. 😈 

**LinkReplacer**: uses a markdown parser to find all links in `.md` files, and replaces links where an appropriate replacement is found. It uses a full markdown parser, as I wanted to be sure I was only replacing actual links - and not literal urls or urls in code samples - which may be intentionally http rather than https.

If the docs team are happy with this approach, I'd like to run this over the rest of the repo, in subsequent PRs. 🙂

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
Thanks again @ChrisMaddock This is really useful and thorough.

I'll approve it now. 

Tagging @mairaw to merge.

File: docs/standard/cross-platform/index.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ You can develop apps for both Microsoft and non-Microsoft platforms by using the
...
+|Share source code between Windows Phone 8.1 and Windows 8.1 apps|**Shared projects** (Universal Apps template in Visual Studio 2013, Update 2).<br /><br /> -   Currently no Visual Basic support.<br />-   You can separate platform-specific code by using #`if` statements.<br /><br /> For details, see:<br /><br /> -   [Build apps that target Windows and Windows Phone by using Visual Studio](https://msdn.microsoft.com/library/windows/apps/dn609832.aspx) (MSDN article)<br />-   [Using Visual Studio to build Universal XAML Apps](http://blogs.msdn.com/b/visualstudio/archive/2014/04/14/using-visual-studio-to-build-universal-xaml-apps.aspx) (blog post)<br />-   [Using Visual Studio to Build XAML Converged Apps](https://channel9.msdn.com/Events/Build/2014/3-591) (video)|
```
Comment (mairaw) [inline docs/standard/cross-platform/index.md]: can you update this http link to be https://blogs.msdn.microsoft.com/visualstudio/2014/04/14/using-visual-studio-to-build-universal-xaml-apps/?

File: docs/standard/cross-platform/index.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ You can develop apps for both Microsoft and non-Microsoft platforms by using the
...
...t varies by platform.<br />-   Portable Class Library projects that target Windows 8.1 and Windows Phone 8.1 support Windows Runtime APIs and XAML. These features aren't available in older versions of the Portable Class Library.<br />-   If needed, you can abstract out platform-specific code by using interfaces or abstract classes.<br /><br /> For details, see:<br /><br /> -   [Portable Class Library](../../../docs/standard/cross-platform/cross-platform-development-with-the-portable-class-library.md) (MSDN article)<br />-   [How to Make Portable Class Libraries Work for You](http://blogs.msdn.com/b/dsplaisted/archive/2012/08/27/how-to-make-portable-class-libraries-work-for-you.aspx) (blog post)<br />-   [Using Portable Class Library with MVVM](../../../docs/standard/cross-platform/using-portable-class-library-with-model-view-view-model.md) (MSDN article)<br />-   [App Resources for Libraries That Target Multiple Platforms](../../../docs/standard/cross-platform/app-resources-for-libraries-that-target-multiple-platforms.md) (MSDN article)<br />-   [.NET Portability Analyzer](http://visualstudiogallery.msdn.microsoft.com/1177943e-cfb7-4822-a8a6-e56c7905292b) (Visual Studio extension)|
```
Comment (mairaw) [inline docs/standard/cross-platform/index.md]: another one that can be changed to https if new url is used: https://blogs.msdn.microsoft.com/dsplaisted/2012/08/27/how-to-make-portable-class-libraries-work-for-you/

**Review by mairaw — CHANGES_REQUESTED** [review]
@ChrisMaddock this looks awesome! Thank you so much! Left two comments for two other URLs that could be replaced as well.

**Comment by ChrisMaddock** [comment]
Thanks @mairaw - changes made. 

I had a separate plan to tackle some of the links which could be redirected to a https site at a different URL - I'd done an enhancement to this tool which finds all the links that redirect to `docs.microsoft`, and replaces them with the appropriate relative link. Sounds like we could do the same for `blogs.msdn.microsoft.com`. 😄

**Review by mairaw — APPROVED** [review]
Thank you for addressing my comments @ChrisMaddock. Yes, adding a special case for blogs seems like a great idea.

**Pull request merged**